virtme/guest/virtme-sshd-script

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+#
+# Initialize ssh server for remote connections (option `--server ssh`)
+
+if [ ! -n "${virtme_ssh_user}" ]; then
+    echo "ssh: virtme_ssh_user is not defined" >&2
+    exit 1
+fi
+
+# Generate authorized_keys adding the host's user public keys, but only if
+# the rootfs is mounted in read-only.
+SSH_HOME=$(getent passwd "${virtme_ssh_user}" | cut -d: -f6)
+if [ ! -e "${SSH_HOME}" ]; then
+    # Setup an arbitrary ssh location, just to be able to start sshd.
+    SSH_HOME=/run/ssh
+fi
+if grep ' / ' /proc/mounts | grep -q ' ro,'; then
+    cat ${SSH_HOME}/.ssh/id_*.pub >> "${SSH_HOME}/.ssh/authorized_keys" 2>/dev/null
```

Comment:
I guess `SSH_HOME` will not contain spaces. Just in case this patch fixes warnings reported by shellcheck:

```diff
diff --git a/virtme/guest/virtme-sshd-script b/virtme/guest/virtme-sshd-script
index 334ae24..5ef2915 100755
--- a/virtme/guest/virtme-sshd-script
+++ b/virtme/guest/virtme-sshd-script
@@ -2,7 +2,7 @@
 #
 # Initialize ssh server for remote connections (option `--server ssh`)
 
-if [ ! -n "${virtme_ssh_user}" ]; then
+if [ -z "${virtme_ssh_user}" ]; then
     echo "ssh: virtme_ssh_user is not defined" >&2
     exit 1
 fi
@@ -15,7 +15,7 @@ if [ ! -e "${SSH_HOME}" ]; then
     SSH_HOME=/run/ssh
 fi
 if grep ' / ' /proc/mounts | grep -q ' ro,'; then
-    cat ${SSH_HOME}/.ssh/id_*.pub >> "${SSH_HOME}/.ssh/authorized_keys" 2>/dev/null
+    cat "${SSH_HOME}"/.ssh/id_*.pub >> "${SSH_HOME}/.ssh/authorized_keys" 2>/dev/null
     chown "${virtme_ssh_user}" "${SSH_HOME}/.ssh/authorized_keys" 2>/dev/null
 fi
 
@@ -23,7 +23,10 @@ fi
 CACHE_DIR=${SSH_HOME}/.cache/virtme-ng/.ssh
 mkdir -p "${CACHE_DIR}/etc/ssh"
 ssh-keygen -A -f "${CACHE_DIR}"
-ARGS=(); for i in "${CACHE_DIR}"/etc/ssh/ssh_host_*_key; do ARGS+=(-h ${i}); done
+ARGS=()
+for key in "${CACHE_DIR}"/etc/ssh/ssh_host_*_key; do
+    ARGS+=(-h "${key}")
+done
 
 # Start sshd.
 mkdir -p /run/sshd
```

(while at it, I split the for-loop on different lines ; shellcheck was only asking to add double quotes around `${i}`)